packages/pipeline/src/Counter.tsx

```diff
@@ -0,0 +1,158 @@
+   */
+  size: Size;
+
+  /**
+   * Alter the visual apperance of the component. Inheritted from the parent Pipeline component.
+   */
+  variant: Variant;
+}
+
+const getBaseStyle = ({
+  size = Size.XSmall,
+  variant = Variant.Default,
+}: StateForStyles): string => {
+  const { chevron, fontSize, fontWeight, gutter, height, lineHeight } = layout[
+    size
+  ];
+
+  const { color, secondary } = colors[variant];
+  const outerSize = height / 2;
+  const offset = outerSize + chevron.size * 2;
+
+  return cx(
+    getRootStyle({ size, variant }),
+    getChildStyle({ size, variant }),
+    css`
+      background-color: ${secondary.backgroundColor};
+      color: ${color};
+      padding: ${gutter.vertical}px ${gutter.horizontal}px;
+      margin-right: ${offset}px;
+      font-size: ${fontSize}px;
+      font-weight: ${fontWeight};
+      line-height: ${lineHeight};
+      flex: 1 1 auto;
+      white-space: nowrap;
+      z-index: 2;
+
+      &::before {
+        white-space: nowrap;
+        content: '+' counter(hiddenCount);
+      }
+    `,
+  );
+};
+
+const getCounterChevronStyle = ({
+  size = Size.XSmall,
+  variant = Variant.Default,
+}: StateForStyles): string => {
+  const { chevron, height } = layout[size];
+  const { secondary } = colors[variant];
+  const outerSize = height / 2;
+
+  return cx(
+    getChevronStyle({ size, variant }),
+    css`
+      &::before {
+        background-color: ${secondary.backgroundColor};
+        box-shadow: 0 0 0 ${chevron.size}px transparent,
+          0 0 0 ${outerSize}px transparent;
+      }
+    `,
+  );
+};
+
+const getStatefulStyles = (state: StateForStyles) => ({
+  base: getBaseStyle(state),
+  chevron: getCounterChevronStyle(state),
+});
+
+/**
+ * # Counter
+ *
+ * React Component to render the counter for the number of hidden stages in the Pipeline component.
+ *
+ * ```
+ * <Counter />
+ * ```
+ * @param props.className Classname applied to Counter content container.
+ */
+const Counter = forwardRef(
+  (
+    { className, children, size, variant, ...rest }: CounterProps,
+    ref: Ref<HTMLDivElement>,
+  ): ReactElement => {
+    const { base: baseStyle, chevron: chevronStyle } = getStatefulStyles({
+      size,
+      variant,
+    });
+
+    return (
+      <div
+        {...rest}
+        {...counterAttr.prop}
+        data-testid="pipeline-counter"
+        className={cx(baseStyle, className)}
+        ref={ref}
+      >
+        {/* Children will be the tooltip provided by the Pipeline component */}
+        {children}
+
+        <div
+          {...counterChevronAttr.prop}
+          data-testid="pipeline-counter-chevron"
+          className={chevronStyle}
+        />
+      </div>
+    );
+  },
+);
+
+Counter.displayName = 'Counter';
+
+Counter.propTypes = {
+  children: PropTypes.node,
+  className: PropTypes.string,
+  size: PropTypes.oneOf(Object.values(Size)).isRequired,
+  variant: PropTypes.oneOf(Object.values(Variant)).isRequired,
+};
+
+Counter.defaultProps = {
```

Comment:
We're trying to not use `defaultProps` as the React core team has mentioned they intend to deprecate them in the future in favor of destructuring defaults. Instead we're leveraging ES6 so we could set the default value in the function arguemnt.

Ditto for all other defaultProps usages.

Reply:
Not a problem, I wasn't sure whether the team followed this pattern or not. I think it would be valuable to add this to a set of design patterns to be added to the contributing guidelines. I suggested adding this to the contributing guidelines that I submitted to the LeafyGreen team.

Have you seen this document? I can't recall if you were on the shared invites?

Reply:
Resolved by ed26972

Reply:
I saw the PR get merged with it? I think it's great!

And yes this would be great to add to that list. Feel free to include that in this PR if you like.